### PR TITLE
week 3 assessment

### PR DESCRIPTION
**Pull request by Phalco1**
# Submitting Week 3 Assessment

### ✅ Completed Files

Select the following as confirmation that the files in this repository are complete.

- [X ] Coding Challenges
- [X ] Coding Concepts
- [ X] Interview Questions

### ✋ Reminder

Do NOT merge your assessment!

### 📝 Feedback

If you would like specific feedback please leave details here:

### CODE REVIEW

File: code-challenges.test.js [hunk 1]
```diff
@@ -14,20 +14,60 @@
...
+// 1.) Declare a fuction to set up the fibonacci sequence. Create a function that has an array with 0 and 1.
 
-// a) Create a test with expect statements for each of the variables provided.
+function fibonacci(n) {
+    const fib = [1, 1]
+
+// 2.) Make a for loop to iterate on the array.
+  for (let i = 0; i< n - 2; i++) {
+
+    // 3.) Have fib identify the index in the array and have it equal to the previous 2 indexes.
```
Comment (chrisyhlee) [inline code-challenges.test.js:45]: Your pseudo code says the index is equal to the previous 2, but the way your code is set up, it does not accurately describe what is going on. In your code, you are starting iteration on index 0, and saying that 2 indexes after the current one is equal to the current index plus the next index. I suggest rewording this to make it more accurate.

File: code-challenges.test.js [hunk 2]
```diff
@@ -14,20 +14,60 @@
...
+describe("leastToGreatest", () => {
+  it("Create a test with expect statements for each of the variables provided.", () => {
+      expect(dayArray(studyMinutesWeek1).toEqual([15, 15, 20, 30, 30, 60, 90]))
+      expect(dayArray(studyMinutesWeek2).toEqual([10, 15, 20, 45, 60, 65, 100]))
+  })
+})
+
+// TypeError: dayArray(...).toEqual is not a function
+
+// It's giving me this error code and i am unable to detect the problem. To my knowledge the function i wrote works and i feel like I also wrote the test right so im confused on what i did wrong.
```
Comment (chrisyhlee) [inline code-challenges.test.js:70]: At the end of lines 63 and 64, one of those closing parentheses should be before the `.toEqual`, to close the parentheses for `expect()`

File: code-challenges.test.js [hunk 3]
```diff
@@ -51,14 +91,32 @@ const studyMinutesWeek2 = {
...
+
+const dayArray = (object) => {
+  return Object.values(object).sort((a, b) => (a - b) )
+}
+// console.log(dayArray(studyMinutesWeek1))
+// console.log(dayArray(studyMinutesWeek2))
 
 // Pseudo code:
 
+// 1.) Declare a function that takes the value of the object and displays the value in the array.
```
Comment (chrisyhlee) [inline code-challenges.test.js:106]: Definitely need more explanation here, you basically only have your input and output. You should include that you are using Object.values to return an array of the values of the object, then using sort to sort from least to greatest.

File: code-challenges.test.js [hunk 4]
```diff
@@ -71,3 +129,26 @@ const accountTransactions3 = []
...
+const accumSum = (array) => {
+// 2.) Create a for loop where the sum, that starts at 0, is added to the elements and returns the new sum into the empty array. The for loop will iterate over the length of the array. 
+  let sum = 0 
+  let newSum = []
+  for (let i = 0; i<array.length; i++) {
+    
+// 3.) Create the logic in the for loop by having the sum, 0, plus the value of the element to equal a sum. Then to take that value and push it to the value in the next index and add those values together.
+
+    sum = array[i] + sum 
+    array[i] = array[i] + array[i]
```
Comment (chrisyhlee) [inline code-challenges.test.js:144]: This line (144) is not necessary and can be removed entirely.

File: code-challenges.test.js [hunk 5]
```diff
@@ -71,3 +129,26 @@ const accountTransactions3 = []
...
+
+// 1.) Create a function that takes in an array (parameter) and takes in any array 
+
+const accumSum = (array) => {
+// 2.) Create a for loop where the sum, that starts at 0, is added to the elements and returns the new sum into the empty array. The for loop will iterate over the length of the array. 
+  let sum = 0 
+  let newSum = []
+  for (let i = 0; i<array.length; i++) {
+    
+// 3.) Create the logic in the for loop by having the sum, 0, plus the value of the element to equal a sum. Then to take that value and push it to the value in the next index and add those values together.
```
Comment (chrisyhlee) [inline code-challenges.test.js:141]: Confusing pseudo code here. An example to make it clear would be:
- update sum per iteration to equal current sum plus value at current index
- push the updated sum into newSum array

File: interview-questions.md [hunk 6]
```diff
@@ -8,33 +8,33 @@ Then, research the question to expand on your answer. Even if you feel you have
...
-Researched answer:
+Researched answer: State values are local or internal and have to be coded by the developer. Props are external and can only be read by parent file.
 
 5. What is the DOM?
 
-Your answer:
+Your answer: The interface that you write the code. Digital Object Model?
 
-Researched answer:
+Researched answer: Document Object Model. It represents the entire user interface of the web app as a tree data structure.
```
Comment (chrisyhlee) [inline interview-questions.md:37]: These interview questions need more elaboration. They are very to the point, but that's not necessarily what you want in an actual interview. As they currently are, they leave a lot of room for the interviewer to doubt if you actually know what you're talking about or not.

**Review by chrisyhlee — APPROVED** [review]
Paul, good effort on week 3 assessment, officially halfway through assessments. This currently displays Novice level. I suggest taking time to really make sure you understand the tools and logic you find in your research so that you can give a clear explanation in your pseudo code. Your pseudo code should be the main focus for this week. I left you feedback throughout the assessment as well, so read through those and let me know if you have further questions. Keep up the hard work, you got this.

**Comment by Phalco1** [comment]
Thank you for your feedback. I find myself having to think out loud and I'm unable to translate that well to pseudo code. After I think out loud I usually get stuck writing the pseudo code and skip it and attempt to write the function from my thoughts and theories; and then try to make my pseudo code make sense to my function. That's what i find myself doing and it's frustrating because that's not what you guys are teaching or advising us to do. So I just keep studying when I have the time until things start to click more and my process is in line with what LEARN is teaching. Is there any advice that can help me with what you see in my 3 assessments? Is there a pattern that I'm not seeing?

**Comment by chrisyhlee** [comment]
That is 100% normal. In the beginning, almost everyone does exactly what you're saying: make the code first then pseudo code later. That's just because you're not at the point of planning out what tools and concepts you need beforehand in order to get to your solution. It will come with practice and time. My biggest suggestion is to review and study any material you don't feel 100% confident on that you have been taught in class so far. You can also go back to previous assessments and redo code challenges with a different approach. Eventually, you will find yourself being able to plan out your whole function ahead of time. You're doing great!! Don't get frustrated over this, I went through the same thing myself!
